src/app/models/mod.rs

```diff
@@ -0,0 +1,181 @@
+use mongodb::bson::{oid::ObjectId, DateTime};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct User {
+    #[serde(rename = "_id", skip_serializing_if = "Option::is_none")]
+    id: Option<ObjectId>,
+    first_name: String,
+    last_name: String,
+    national_health_identifer: String,
+    email_address: String,
+    hashed_password: String,
+    is_patient: bool,
+    caregivers: Vec<ObjectId>,
+    form_templates: Vec<Form>,
+    form_filled: Vec<Form>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct Form {
+    #[serde(rename = "_id", skip_serializing_if = "Option::is_none")]
+    id: Option<ObjectId>,
+    title: String,
+    created_by: ObjectId,
+    created_at: DateTime,
+    questions: Vec<Question>,
+    events: Vec<Event>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub enum Event {
+    FormSubmitted(FormSubmitted),
+    QuestionEdited(QuestionEdited),
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct QuestionEdited {
+    question_id: ObjectId,
+    former_question: Question,
+    new_question: Question,
+    edited_by: ObjectId,
+    edited_at: DateTime,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct FormSubmitted {
+    answers: Vec<QuestionAndAnswer>,
+    submitted_by: ObjectId,
+    submitted_at: DateTime,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub enum Question {
+    Multichoice(MultichoiceQuestion),
+    Slider(SliderQuestion),
+    FreeForm(FreeFormQuestion),
+}
+
+pub type MultichoiceAnswer = ObjectId;
+pub type SliderAnswer = f64;
+pub type FreeFormAnswer = String;
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub enum QuestionAndAnswer {
+    Multichoice(MultichoiceQuestion, MultichoiceAnswer),
+    Slider(SliderQuestion, SliderAnswer),
+    FreeForm(FreeFormQuestion, FreeFormAnswer),
+}
+
+// In the template, the answer will be set to the default
+// it is when the form is opened
+#[derive(Serialize, Deserialize, Clone, Debug, Default)]
+pub struct FreeFormQuestion {
+    id: ObjectId,
```

Comment:
Should be serde-renamed to `_id`, assuming that we need a unique id at all. Same goes for `SliderQuestion`, `MultichoiceQuestion`, and `MultichoiceQuestionOption`.